sky/serve/service_spec.py

```diff
@@ -60,17 +68,46 @@ def __init__(
                     'Currently, SkyServe will cleanup failed replicas'
                     'and auto restart it to keep the service running.')
 
-        self._readiness_path = readiness_path
-        self._initial_delay_seconds = initial_delay_seconds
-        self._min_replicas = min_replicas
-        self._max_replicas = max_replicas
-        self._target_qps_per_replica = target_qps_per_replica
-        self._post_data = post_data
+        if spot_policy is not None:
+            if autoscaler is not None or spot_placer is not None:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        'Cannot specify `autoscaler` and `spot_placer`'
+                        'when `spot_policy` is specified.')
+            # TODO(MaoZiming): do not hardcode the name
+            if spot_policy == 'SpotHedge':
+                autoscaler = 'SpotOnDemandRequestRateAutoscaler'
+                spot_placer = 'DynamicFailover'
+            else:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(f'Unsupported spot policy: {spot_policy}.')
+
+        if autoscaler not in autoscalers.Autoscaler.get_autoscaler_names():
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(f'Unsupported autoscaler: {autoscaler}.')
 
-        self._upscale_delay_seconds = (
+        if (spot_placer is not None and
+                spot_placer not in spot_policies.SpotPlacer.get_policy_names()):
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(f'Unsupported spot placer: {spot_placer}.')
+
+        self._readiness_path: str = readiness_path
+        self._initial_delay_seconds: int = initial_delay_seconds
+        self._min_replicas: int = min_replicas
+        self._max_replicas: Optional[int] = max_replicas
+        self._target_qps_per_replica: Optional[float] = target_qps_per_replica
+        self._post_data: Optional[Dict[str, Any]] = post_data
+        self._spot_placer: Optional[str] = spot_placer
+        self._autoscaler: str = autoscaler
+        self._num_overprovision: Optional[int] = num_overprovision
+        self._init_replicas: Optional[int] = init_replicas
+        self._extra_on_demand_replicas: Optional[int] = extra_on_demand_replicas
+        # _spot_zones will be set by set_spot_zones from resource config.
+        self._spot_zones: Optional[List[str]] = None
```

Comment:
Do we still need this variable? Could we just use the information in the task yaml?

Reply:
this is just a shortcut to store all the zones; so that we don't have to fill the `launchable_zones` again.

Reply:
I am still feeling like it adds complexity for the code. Though this issue is related to which field we want to enable in multi resources case; we could defer the decision later